tests/i18n/test_extraction.py

```diff
@@ -687,6 +687,25 @@ def test_all_locales(self):
         self.assertTrue(os.path.exists(self.PO_FILE_KO))
         self.assertFalse(os.path.exists('locale/_do_not_pick/LC_MESSAGES/django.po'))
 
+    def test_locales_format_valid(self):
+        """Check the standard output(stdout) message in case the format of locale is valid"""
+        out = StringIO()
+        LOCALE = ['en_IN', 'th_TH_TH', 'asa_TZ', 'az_Cyrl', 'yue_Hant_HK', 'haw_US']
+        for locale in LOCALE:
+            with self.subTest(locale):
+                management.call_command("makemessages", locale=[locale], verbosity=1, stdout=out)
+                self.assertNotIn("invalid", out.getvalue())
+                self.assertIn("processing locale %s" % locale, out.getvalue())
+
+    def test_locales_format_invalid(self):
+        """Check the standard output(stdout) message in case the format of locale is invalid"""
+        out = StringIO()
+        LOCALE = ['en-IN', 'th_TH_in', 'aSA_tZ', 'aZ_CyRl', 'EN-IN', 'EN-in']
+        for locale in LOCALE:
+            with self.subTest(locale):
+                management.call_command("makemessages", locale=[locale], stdout=out)
+                self.assertIn("invalid locale %s" % locale, out.getvalue())
```

Comment:
I would test `self.assertNotIn('processing locale %s...` to capture the case I mentioned above where we want to `continue`.